pgsqltoolsservice/connection/connection_service.py

```diff
@@ -300,16 +294,25 @@ def _close_connections(connection_info: ConnectionInfo, connection_type=None):
                 # Ignore errors when disconnecting
                 pass
         return True
+    
+    def _get_provider_name(self, owner_uri):
```

Comment:
Couple of things here:
1. The method states get.. but we are also setting a property of the class
2.  Pgtoolsservice would be loaded for a single type. So could we expect ProviderName in the startup args ?

Reply:
I completely agree, so I added the provider as part of the start-up args. Now the ServiceProvider will have a provider_name when it is instantiated, and the services can access it through a public property.